// [[Reason: checker.typeToTypeNode deliberately fails on types that originate from node_modules.]] ////

//// [tests/cases/conformance/node/nodeModulesExportsSpecifierGenerationPattern.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,24 +1,28 @@
 
 
 //// [index.d.ts]
-export declare const a: import("inner/other.js").Thing;
+export declare const a: invalid;
 //# sourceMappingURL=index.d.ts.map
 /// [Errors] ////
 
 error TS2468: Cannot find global value 'Promise'.
 index.ts(2,23): error TS2307: Cannot find module 'inner/other' or its corresponding type declarations.
+index.ts(3,14): error TS9010: Variable must have an explicit type annotation with --isolatedDeclarations.
 index.ts(3,19): error TS1378: Top-level 'await' expressions are only allowed when the 'module' option is set to 'es2022', 'esnext', 'system', 'node16', or 'nodenext', and the 'target' option is set to 'es2017' or higher.
 index.ts(3,25): error TS2712: A dynamic import call in ES5/ES3 requires the 'Promise' constructor.  Make sure you have a declaration for the 'Promise' constructor or include 'ES2015' in your '--lib' option.
 
 
 !!! error TS2468: Cannot find global value 'Promise'.
-==== index.ts (3 errors) ====
+==== index.ts (4 errors) ====
     // esm format file
     import { Thing } from "inner/other";
                           ~~~~~~~~~~~~~
 !!! error TS2307: Cannot find module 'inner/other' or its corresponding type declarations.
     export const a = (await import("inner/index.js")).x();
+                 ~
+!!! error TS9010: Variable must have an explicit type annotation with --isolatedDeclarations.
+!!! related TS9028 index.ts:3:14: Add a type annotation to the variable a.
                       ~~~~~
 !!! error TS1378: Top-level 'await' expressions are only allowed when the 'module' option is set to 'es2022', 'esnext', 'system', 'node16', or 'nodenext', and the 'target' option is set to 'es2017' or higher.
                             ~~~~~~~~~~~~~~~~~~~~~~~~
 !!! error TS2712: A dynamic import call in ES5/ES3 requires the 'Promise' constructor.  Make sure you have a declaration for the 'Promise' constructor or include 'ES2015' in your '--lib' option.
